crates/cairo-addons/src/vm/hint_definitions/utils.rs

```diff
@@ -0,0 +1,145 @@
+use std::collections::HashMap;
+
+use cairo_vm::{
+    hint_processor::{
+        builtin_hint_processor::hint_utils::{
+            get_maybe_relocatable_from_var_name, get_ptr_from_var_name, insert_value_from_var_name,
+        },
+        hint_processor_definition::HintReference,
+    },
+    serde::deserialize_program::ApTracking,
+    types::{exec_scope::ExecutionScopes, relocatable::MaybeRelocatable},
+    vm::{
+        errors::{hint_errors::HintError, memory_errors::MemoryError},
+        vm_core::VirtualMachine,
+    },
+    Felt252,
+};
+
+use crate::vm::hints::Hint;
+
+#[allow(non_snake_case)]
+pub fn bytes__eq__() -> Hint {
```

Comment:
I think this could be refactored
The same thing is done for self and other and then they are compared and the same behavior is done if there is a diff or not.
Could be factorized with helper closures ?

Reply:
yes excellent idea!